tests/sflow/test_sflow.py

```diff
@@ -56,13 +58,22 @@ def setup(duthosts, rand_one_dut_hostname, ptfhost, tbinfo, config_sflow_feature
 
     config_dut_ports(duthost, var['test_ports'][0:2], vlan=1000)
 
-    for port_channel, interfaces in list(mg_facts['minigraph_portchannels'].items()):
-        for port in interfaces['members']:
-            var['sflow_ports'][port] = {}
-            var['sflow_ports'][port]['ifindex'] = get_ifindex(duthost, port)
-            var['sflow_ports'][port]['port_index'] = get_port_index(duthost, port)
-            var['sflow_ports'][port]['ptf_indices'] = mg_facts['minigraph_ptf_indices'][port]
-            var['sflow_ports'][port]['sample_rate'] = SFLOW_RATE_DEFAULT
+    upstream_ports = []
+    if len(mg_facts["minigraph_portchannels"]) == 0:
+        topo = tbinfo["topo"]["type"]
+        upstream_neigh_type = get_upstream_neigh_type(topo)
+        upstream_ports = get_neighbor_port_list(duthost, upstream_neigh_type)
```

Comment:
We could add an assert if len(upstream_ports) is 0

Reply:
done